packages/lib/src/components/internal/Address/components/AddressSearch.tsx

```diff
@@ -0,0 +1,98 @@
+import Field from '../../FormFields/Field';
+import { Fragment, h } from 'preact';
+import renderFormField from '../../FormFields';
+import { AddressLookupItem } from '../types';
+import { useCallback, useEffect, useMemo, useState } from 'preact/hooks';
+import './AddressSearch.scss';
+import useCoreContext from '../../../../core/Context/useCoreContext';
+import { debounce } from '../utils';
+
+export type OnAddressLookupType = (
+    value: string,
+    actions: {
+        resolve: (value: Array<AddressLookupItem>) => void;
+        reject: (reason?: any) => void;
+    }
+) => Promise<void>;
+
+interface AddressSearchProps {
+    onAddressLookup?: OnAddressLookupType;
+    onSelect: any; //TODO
+    onManualAddress: any;
+    externalErrorMessage: string;
+    hideManualButton: boolean;
+}
+
+export default function AddressSearch({ onAddressLookup, onSelect, onManualAddress, externalErrorMessage, hideManualButton }: AddressSearchProps) {
+    const [formattedData, setFormattedData] = useState([]);
+    const [originalData, setOriginalData] = useState([]);
+
+    const [errorMessage, setErrorMessage] = useState('');
+
+    const { i18n } = useCoreContext();
+    const mapDataToSelect = data => data.map(({ id, name }) => ({ id, name }));
+
+    const onInput = useCallback(
+        async event => {
+            new Promise<Array<AddressLookupItem>>((resolve, reject) => {
+                onAddressLookup(event, { resolve, reject });
+            })
+                .then(data => {
+                    setOriginalData(data);
+                    setFormattedData(mapDataToSelect(data));
+                    setErrorMessage('');
+                })
+                .catch(reason => {
+                    setErrorMessage(reason);
+                    console.error('error', reason);
+                });
+        },
+        [onAddressLookup]
+    );
+
+    // update error message when there's a new one
+    useEffect(() => {
+        setErrorMessage(externalErrorMessage);
+    }, [externalErrorMessage]);
+
+    const onChange = event => {
+        if (!event.target.value) {
+            setErrorMessage(i18n.get('address.errors.incomplete'));
+            return;
+        }
+        const value = originalData.find(item => item.id === event.target.value);
+        onSelect(value);
+        setFormattedData([]);
+    };
```

Comment:
Out of curiosity - What is the difference between `onInput` and `onChange` ? They seem be the same thing but they are not 😅